cpp/include/raft/neighbors/nn_descent.cuh

```diff
@@ -56,9 +56,9 @@ namespace raft::neighbors::experimental::nn_descent {
  * @return index<IdxT> index containing all-neighbors knn graph in host memory
  */
 template <typename T, typename IdxT = uint32_t>
-index<IdxT> build(raft::resources const& res,
-                  index_params const& params,
-                  raft::device_matrix_view<const T, int64_t, row_major> dataset)
+index<detail::DistData_t, IdxT> build(raft::resources const& res,
```

Comment:
The signature looks incorrect to me. If `T = double` then the index is also storing distances in `double`, right?
```suggestion
index<T, IdxT> build(raft::resources const& res,
```

Reply:
But in`nn_descent.cuh` , the build is declared as `index<DistData_t, IdxT> build`, where `DistData_t` is declared as `float`.
So changing it like this results in not being able to call `build` when the data type is something other that `float`.
Should I change `nn_descent.cuh` so that it can take the distance type as a template parameter?

Reply:
You can just copy the data from float to type T. Unfortunately, you'll not be able to use raft::copy anymore and you'll have to use raft::linalg::unary_op most likely. 

That way, you don't have to change anything in detail/nn_descent.cuh and you can still solve NN Descent in float, and then copy to the expected type T of the index.

Reply:
==Talked about this online and decided to keep it as float==